target/linux/ath79/files/drivers/mtd/nand/rb4xx_nand.c

```diff
@@ -0,0 +1,399 @@
+/*
+ *  NAND flash driver for the MikroTik RouterBoard 4xx series
+ *
+ *  Copyright (C) 2008-2011 Gabor Juhos <juhosg@openwrt.org>
+ *  Copyright (C) 2008 Imre Kaloz <kaloz@openwrt.org>
+ *
+ *  This file was based on the driver for Linux 2.6.22 published by
+ *  MikroTik for their RouterBoard 4xx series devices.
+ *
+ *  This program is free software; you can redistribute it and/or modify it
+ *  under the terms of the GNU General Public License version 2 as published
+ *  by the Free Software Foundation.
+ */
+
+#include <linux/version.h>
+#include <linux/kernel.h>
+#include <linux/module.h>
+#include <linux/init.h>
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4,14,0)
```

Comment:
<4.14 is no longer a target, maybe this can be removed?

Reply:
So I actually have my doubts this rb4xx NAND driver is needed, I have a hunch a generic driver could do the same job.

N.B. This is the hardware I need to support:

```
[    0.713634] Mikrotik RB4xx NAND flash driver version 0.2.1
[    0.730750] nand: device found, Manufacturer ID: 0x20, Chip ID: 0xf1
[    0.737154] nand: ST Micro NAND01GW3B2CN6
[    0.741163] nand: 128 MiB, SLC, erase size: 128 KiB, page size: 2048, OOB size: 64
[    0.748738] Scanning device for bad blocks
[    0.906749] Creating 3 MTD partitions on "NAND01GW3B2CN6":
[    0.912244] 0x000000000000-0x000000040000 : "booter"
[    0.918015] 0x000000040000-0x000000400000 : "kernel"

```